options.go

```diff
@@ -17,80 +17,74 @@
 package edgedb
 
 import (
-	"fmt"
-	"net/url"
-	"strconv"
-	"strings"
+	"time"
 )
 
 // Options for connecting to an EdgeDB server
 type Options struct {
-	Host     string `json:"host"`
-	Port     int    `json:"port"`
-	User     string `json:"user"`
-	Database string `json:"database"`
-	Password string `json:"password"`
-	admin    bool
+	// Hosts is a slice of database host addresses as one of the following
+	//
+	// - an IP address or domain name
+	//
+	// - an absolute path to the directory
+	//   containing the database server Unix-domain socket
+	//   (not supported on Windows)
+	//
+	// If the slice is empty, the following will be tried, in order:
+	//
+	// - host address(es) parsed from the dsn argument
+	//
+	// - the value of the EDGEDB_HOST environment variable
+	//
+	// - on Unix, common directories used for EdgeDB Unix-domain sockets:
+	//   "/run/edgedb" and "/var/run/edgedb"
+	//
+	// - "localhost"
+	Hosts []string
 
-	MaxConns int
-	MinConns int
-}
-
-func (o *Options) network() string {
-	if o.admin {
-		return "unix"
-	}
-	return "tcp"
-}
+	// Ports is a slice of port numbers to connect to at the server host
+	// (or Unix-domain socket file extension).
+	//
+	// Ports may either be:
+	//
+	// - the same length ans Hosts
+	//
+	// - a single port to be used all specified hosts
+	//
+	// - empty indicating the value parsed from the dsn argument
+	//   should be used, or the value of the EDGEDB_PORT environment variable,
+	//   or 5656 if neither is specified.
```

Comment:
Not sure I understand structure here. Is this "Hosts <crossproduct> Ports" or parallel arrays where first host in Hosts corresponds to first port in ports?

What are the downsides of this being an array of pairs?

Reply:
This was an attempt to follow the js driver behavior. The builder pattern would be much a nicer API in go it just didn't seem like that design was finalized. I'm just trying to get off the ground with this PR.